upload/src/addons/SV/WarningImprovements/Finder/WarningCategory.php

```diff
@@ -8,15 +8,15 @@ class WarningCategory extends Finder
 {
     public function inParentWarningCategory(\SV\WarningImprovements\Entity\WarningCategory $warningCategory)
     {
-        $this->where('parent_warning_category_id', $warningCategory->parent_warning_category_id);
+        $this->where('parent_category_id', $warningCategory->parent_category_id);
         $this->order('display_order');
 
         return $this;
     }
 
     public function rootOnly()
     {
-        $this->where('parent_warning_category_id', '=', 0);
+        $this->where('parent_category_id', '=', 0);
```

Comment:
Since parent_category_id is nullable, this should be a check against null, not 0

Reply:
Side note; I really disliked XF1 using 0 rather than null for a non-reference. XF2 at least looks to be moving towards using SQL properly

Reply:
There are still many places where 0 is being used especially in XFMG

Reply:
I'ld prefer this add-on to use null where possible since it is already doing it.